src/main/kotlin/kotlinx/kover/gradle/plugin/tasks/internal/KoverArtifactGenerationTask.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.gradle.plugin.tasks.internal
+
+import kotlinx.kover.gradle.plugin.commons.*
+import org.gradle.api.*
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.*
+import org.gradle.api.tasks.*
+import org.gradle.kotlin.dsl.*
+import java.io.*
+import javax.inject.*
+
+@CacheableTask
+internal open class KoverArtifactGenerationTask : DefaultTask() {
```

Comment:
It would be nice to drop documentation with explanation why we need artifact and what's inside, just like with `KoverAgentJarTask`